src/main/java/net/smartcosmos/dao/relationships/impl/RelationshipPersistenceService.java

```diff
@@ -32,237 +39,285 @@
     private final SearchSpecifications<RelationshipEntity> searchSpecifications = new SearchSpecifications<RelationshipEntity>();
 
     @Autowired
-    public RelationshipPersistenceService(RelationshipRepository RelationshipRepository,
-            ConversionService conversionService) {
+    public RelationshipPersistenceService(
+        RelationshipRepository RelationshipRepository,
+        ConversionService conversionService) {
         this.relationshipRepository = RelationshipRepository;
         this.conversionService = conversionService;
     }
 
     // region Create
-
     @Override
-    public Optional<RelationshipResponse> create(String tenantUrn, RelationshipCreate createRelationship) throws ConstraintViolationException {
-        // TODO: Implement method
-        throw new UnsupportedOperationException("Not implemented yet");
-    }
-
-    // endregion
+    public Optional<RelationshipResponse> create(String tenantUrn, RelationshipCreate createRelationship) {
 
-    // region Delete
+        if (alreadyExists(tenantUrn, createRelationship)) {
+            return Optional.empty();
+        }
+        RelationshipEntity entity = null;
+        try {
+            UUID tenantId = UuidUtil.getUuidFromUrn(tenantUrn);
+            entity = conversionService.convert(createRelationship, RelationshipEntity.class);
+            entity.setTenantId(tenantId);
+            entity = persist(entity);
+
+        } catch (Exception e) {
+            exceptionLogger("RelationshipPersistenceService:create", e, tenantUrn, createRelationship);
+            throw e;
+        }
+        return Optional.ofNullable(conversionService.convert(entity, RelationshipResponse.class));
+    }
+    //endregion
 
+    // region delete
     @Override
```

Comment:
This would be more efficient and cleaner written like the following.  The way you have it here the code creates an `ArrayList<>` to simply throw it away.
```
@Override
    public List<RelationshipResponse> delete(String tenantUrn, String urn) {

        try {
            UUID tenantId = UuidUtil.getUuidFromUrn(tenantUrn);
            UUID uuid = UuidUtil.getUuidFromUrn(urn);
            return convertList( relationshipRepository.deleteByTenantIdAndId(tenantId, uuid));

        } catch (Exception e) {
            exceptionLogger("RelationshipPersistenceService:delete", e, "tenantUrn: " + tenantUrn, "urn: " + urn);
        }
        return convertList(Collections.emptyList());
    }

````